### ISSUE TITLE
Recommended over C++ yet?

### ISSUE THREAD

**Issue by adamski** [mdaffin/teensy-3-rust#4]
Given your blog post is almost a year old now, wondering if Rust seems stable enough to start a Teensy based project with it?

**Comment by mdaffin** [comment]
Rist has improved alot since I wrote that blog post but you still require the nightly version and will for the foreseeable future. It still lacks dedicated libraries but the core itself has not changed a huge amount; since I wrote the blog I have only had to update it a couple of times due to breaking changes.

There is a large amount of intrest growing in pushing embedded rust forward, you might want to take a look at [rust-embedded/rfcs](https://github.com/rust-embedded/rfcs/issues) for the communities effots on building some of the missing components.

You might also want to look at [discovery](https://japaric.github.io/discovery/) and [copper](https://japaric.github.io/copper/) books by japaric.

But to answer your question: it depends on the level of stability/effort you require/want to apply. I don't recommend it for production level project or projects with a timeline just yet, but for hobbyist ones it is beginning to become viable. You will need to manually write a lot of things yourself and will be quite a large learning curve but if thats not an issue it might be a good time to start looking into it otherwise I would stick with c++ for now and keep an eye on the above projects.
